compositor_pipeline/src/pipeline/input/mp4/reader.rs

```diff
@@ -95,73 +94,28 @@ impl<Reader: Read + Seek + Send + 'static> Mp4FileReader<Reader> {
             avc.map(|avc| (id, track, avc))
         })?;
 
-        // sps and pps have to be extracted from the container, interleaved with [0, 0, 0, 1],
-        // concatenated and prepended to the first frame.
-        let sps = avc
-            .avcc
-            .sequence_parameter_sets
-            .iter()
-            .flat_map(|s| [0, 0, 0, 1].iter().chain(s.bytes.iter()));
-
-        let pps = avc
-            .avcc
-            .picture_parameter_sets
-            .iter()
-            .flat_map(|s| [0, 0, 0, 1].iter().chain(s.bytes.iter()));
-
-        let mut sps_and_pps_payload = Some(sps.chain(pps).copied().collect::<Bytes>());
-
-        let length_size = avc.avcc.length_size_minus_one + 1;
-
-        let sample_unpacker = move |sample: mp4::Mp4Sample| {
-            let mut sample_data = sample.bytes.reader();
-            let mut data: BytesMut = Default::default();
-
-            if let Some(first_nal) = sps_and_pps_payload.take() {
-                data.extend_from_slice(&first_nal);
-            }
-
-            // the mp4 sample contains one h264 access unit (possibly more than one NAL).
-            // the NALs are stored as: <length_size bytes long big endian encoded length><the NAL>.
-            // we need to convert this into Annex B, in which NALs are separated by
-            // [0, 0, 0, 1]. `length_size` is at most 4 bytes long.
-            loop {
-                let mut len = [0u8; 4];
-
-                if sample_data
-                    .read_exact(&mut len[4 - length_size as usize..])
-                    .is_err()
-                {
-                    break;
-                }
-
-                let len = u32::from_be_bytes(len);
-
-                let mut nalu = bytes::BytesMut::zeroed(len as usize);
-                sample_data.read_exact(&mut nalu).unwrap();
-
-                data.extend_from_slice(&[0, 0, 0, 1]);
-                data.extend_from_slice(&nalu);
-            }
-
-            data.freeze()
-        };
+        let mut avcc_box = BytesMut::new().writer();
+        if let Err(err) = avc.avcc.write_box(&mut avcc_box) {
+            warn!("Failed to write mp4 extra data: {err}");
+        }
```

Comment:
`spawn_video_decoder_thread` accepts extra data as bytes so I have to reserialize avcc_box and then parse it again in `spawn_video_decoder_thread`. This could be avoided if we'd accept `ExtraData` instead, but that would mean we'd have to expose it and its parse method to the caller.

Reply:
I'm fine with exposing ExtraData publicly by renaming it e.g. FfmpegH264ExtraData or if it's not ffmpeg specific, then use h264 in the name

My understanding is that byte structure is not specific to ffmpeg to but the name extradata is, so use the correct name there